packages/shared/test-suites/RecordingTestMetadata.ts

```diff
@@ -615,6 +625,13 @@ export async function processGroupedTestCases(
                     }
                   }
                 }
+
+                const currentTest = partialTestRecordings[currentTestRecordingIndex];
+                assert(
+                  currentTest?.id === annotation.message.testId,
+                  `Test id should match "test:start" annotation testId`
+                );
```

Comment:
This isn't strictly _necessary_ but it verifies that things don't get out of sync.